File: pulsar-third/pulsar-chrome-devtools/src/main/java/com/github/kklisura/cdt/protocol/events/debugger/Paused.java
package com.github.kklisura.cdt.protocol.events.debugger;

/*-
 * #%L
 * cdt-java-client
 * %%
 * Copyright (C) 2018 - 2019 Kenan Klisura
 * %%
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * #L%
 */

import com.github.kklisura.cdt.protocol.support.annotations.Experimental;
import com.github.kklisura.cdt.protocol.support.annotations.Optional;
import com.github.kklisura.cdt.protocol.types.debugger.CallFrame;
import com.github.kklisura.cdt.protocol.types.runtime.StackTrace;
import com.github.kklisura.cdt.protocol.types.runtime.StackTraceId;
import java.util.List;
import java.util.Map;

/** Fired when the virtual machine stopped on breakpoint or exception or any other stop criteria. */
public class Paused {

  private List<CallFrame> callFrames;

  private PausedReason reason;

  @Optional private Map<String, Object> data;

  @Optional private List<String> hitBreakpoints;

  @Optional private StackTrace asyncStackTrace;

  @Experimental @Optional private StackTraceId asyncStackTraceId;

  @Experimental @Optional private StackTraceId asyncCallStackTraceId;

  /** Call stack the virtual machine stopped on. */
  public List<CallFrame> getCallFrames() {
    return callFrames;
  }

  /** Call stack the virtual machine stopped on. */
  public void setCallFrames(List<CallFrame> callFrames) {
    this.callFrames = callFrames;
  }

  /** Pause reason. */
  public PausedReason getReason() {
    return reason;
  }

  /** Pause reason. */
  public void setReason(PausedReason reason) {
    this.reason = reason;
  }

  /** Object containing break-specific auxiliary properties. */
  public Map<String, Object> getData() {
    return data;
  }

  /** Object containing break-specific auxiliary properties. */
  public void setData(Map<String, Object> data) {
    this.data = data;
  }

  /** Hit breakpoints IDs */
  public List<String> getHitBreakpoints() {
    return hitBreakpoints;
  }

  /** Hit breakpoints IDs */
  public void setHitBreakpoints(List<String> hitBreakpoints) {
    this.hitBreakpoints = hitBreakpoints;
  }

  /** Async stack trace, if any. */
  public StackTrace getAsyncStackTrace() {
    return asyncStackTrace;
  }

  /** Async stack trace, if any. */
  public void setAsyncStackTrace(StackTrace asyncStackTrace) {
    this.asyncStackTrace = asyncStackTrace;
  }

  /** Async stack trace, if any. */
  public StackTraceId getAsyncStackTraceId() {
    return asyncStackTraceId;
  }

  /** Async stack trace, if any. */
  public void setAsyncStackTraceId(StackTraceId asyncStackTraceId) {
    this.asyncStackTraceId = asyncStackTraceId;
  }

  /**
   * Just scheduled async call will have this stack trace as parent stack during async execution.
   * This field is available only after `Debugger.stepInto` call with `breakOnAsynCall` flag.
   */
  public StackTraceId getAsyncCallStackTraceId() {
    return asyncCallStackTraceId;
  }

  /**
   * Just scheduled async call will have this stack trace as parent stack during async execution.
   * This field is available only after `Debugger.stepInto` call with `breakOnAsynCall` flag.
   */
  public void setAsyncCallStackTraceId(StackTraceId asyncCallStackTraceId) {
    this.asyncCallStackTraceId = asyncCallStackTraceId;
  }
}

File: pulsar-core/pulsar-tools/pulsar-browser/src/main/kotlin/ai/platon/cdt/kt/protocol/events/debugger/Paused.kt
@file:Suppress("unused")
package ai.platon.cdt.kt.protocol.events.debugger

import ai.platon.cdt.kt.protocol.support.annotations.Experimental
import ai.platon.cdt.kt.protocol.support.annotations.Optional
import ai.platon.cdt.kt.protocol.types.debugger.CallFrame
import ai.platon.cdt.kt.protocol.types.debugger.PausedReason
import ai.platon.cdt.kt.protocol.types.runtime.StackTrace
import ai.platon.cdt.kt.protocol.types.runtime.StackTraceId
import com.fasterxml.jackson.`annotation`.JsonProperty
import kotlin.Any
import kotlin.Deprecated
import kotlin.String
import kotlin.collections.List
import kotlin.collections.Map

/**
 * Fired when the virtual machine stopped on breakpoint or exception or any other stop criteria.
 */
data class Paused(
  @param:JsonProperty("callFrames")
  val callFrames: List<CallFrame>,
  @param:JsonProperty("reason")
  val reason: PausedReason,
  @param:JsonProperty("data")
  @param:Optional
  val `data`: Map<String, Any?>? = null,
  @param:JsonProperty("hitBreakpoints")
  @param:Optional
  val hitBreakpoints: List<String>? = null,
  @param:JsonProperty("asyncStackTrace")
  @param:Optional
  val asyncStackTrace: StackTrace? = null,
  @param:JsonProperty("asyncStackTraceId")
  @param:Optional
  @param:Experimental
  val asyncStackTraceId: StackTraceId? = null,
  @param:JsonProperty("asyncCallStackTraceId")
  @param:Optional
  @Deprecated("Deprecated by protocol")
  @param:Experimental
  val asyncCallStackTraceId: StackTraceId? = null,
)
